pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -27,100 +29,164 @@ import (
+// - setting the `completionTime` when all MasterUserRecord have been processed
+func (r *NSTemplateTierReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	logger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
-	logger.Info("Reconciling NSTemplateTier")
 
-	// Fetch the NSTemplateTier tier
+	// fetch the NSTemplateTier tier
 	tier := &toolchainv1alpha1.NSTemplateTier{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, tier)
-	if err != nil {
+	if err := r.client.Get(context.TODO(), request.NamespacedName, tier); err != nil {
 		if errors.IsNotFound(err) {
-			log.Info("NSTemplateTier not found")
+			logger.Info("NSTemplateTier not found")
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
 		logger.Error(err, "unable to get the current NSTemplateTier tier")
 		return reconcile.Result{}, err
 	}
-	if err := r.ensureTemplateUpdateRequest(logger, tier); err != nil {
+
+	// create a new entry in the `status.history`
+	if added, err := r.ensureStatusUpdateRecord(logger, tier); err != nil {
+		logger.Error(err, "unable to insert a new entry in status.updates after NSTemplateTier changed.")
+		return reconcile.Result{}, err
+	} else if added {
+		logger.Info("Requeing after adding a new entry in tier.status.updates")
+		return reconcile.Result{Requeue: true}, nil
+	}
+	if done, err := r.ensureTemplateUpdateRequest(logger, tier); err != nil {
 		logger.Error(err, "unable to ensure TemplateRequestUpdate resource after NSTemplateTier changed.")
 		return reconcile.Result{}, err
+	} else if done {
+		log.Info("No MasterUserRecord to update. Setting the completion timestamp immediately")
+		if err := r.markUpdateRecordAsCompleted(tier); err != nil {
+			logger.Error(err, "unable to mark latest status.update as complete.")
+			return reconcile.Result{}, err
+		}
 	}
 	return reconcile.Result{}, nil
 }
 
+// ensureStatusUpdateRecord adds a new entry in the `status.updates` with the current date/time
+// if needed and cleans-up the previous one if needed (ie, clears the `failedAccounts` array)
+// returns `true` if an entry was added, `err` if something wrong happened
+func (r *NSTemplateTierReconciler) ensureStatusUpdateRecord(logger logr.Logger, tier *toolchainv1alpha1.NSTemplateTier) (bool, error) {
+	hash, err := ComputeHashForNSTemplateTier(*tier)
+	if err != nil {
+		return false, errs.Wrapf(err, "unable to append an entry in the `status.updates` for NSTemplateTier '%s'", tier.Name)
+	}
+	// if there was no previous status:
+	if len(tier.Status.Updates) == 0 {
+		tier.Status.Updates = append(tier.Status.Updates, toolchainv1alpha1.NSTemplateTierHistory{
+			StartTime: metav1.NewTime(time.Now()),
+			Hash:      hash,
+		})
+		return true, r.client.Status().Update(context.TODO(), tier)
+	}
+	// check that last entry
+	if tier.Status.Updates[len(tier.Status.Updates)-1].Hash == hash {
+		logger.Info("current update still in progress")
```

Comment:
And here too..
```suggestion
		logger.Info("current update/check if updated need is still in progress")
```

Reply:
see 80d6ee7: using a slightly different sentence, which is easier to read, at least to me ;)